src/TSTransformer/nodes/transformOptionalChain.ts

```diff
@@ -257,10 +259,12 @@ function transformOptionalChainInner(
 			if (item.callOptional) {
 				if (item.kind === OptionalChainItemKind.PropertyCall) {
 					baseExpression = luau.property(convertToIndexableExpression(baseExpression), item.name);
-				} else {
+				} else if (item.kind === OptionalChainItemKind.ElementCall) {
```

Comment:
Why did you change the else -> else if?

Reply:
i didnt know that it would be narrowed enough; ill remove it when i get back to my computer